test/integrationConfiguration.test.js

```diff
@@ -0,0 +1,217 @@
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+/* eslint mocha/no-mocha-arrows: "off" */
+
+'use strict';
+
+const assert = require('assert');
+const path = require('path');
+
+const IntegrationConfiguration = require('../lib/integrationConfiguration.js');
+
+const plainTextFile = path.join(process.cwd(), "test", "resources/plain-text-file.txt");
+const yamlFile = path.join(process.cwd(), "test", "resources/integration.yaml");
+const jsonFile = path.join(process.cwd(), "test", "resources/integration.json");
+const privateKeyFile = path.join(process.cwd(), "test", "resources/privatekey.pem");
+
+describe('integrationConfiguration.js tests', () => {
+
+    const expected = {
+        metascopes: ['Meta Scope 1', 'Meta Scope 2', 'Meta Scope 3'],
+        technicalAccount: {
+            id: 'Tech Id',
+            org: 'IMS Org Id',
+            clientId: 'Client Id',
+            clientSecret: 'Client Secret',
+            privateKey: '-----BEGIN PRIVATE KEY-----\n-----END PRIVATE KEY-----\n'
+        }
+    };
+
+    it('Input file does not exist', async () => {
+
+        let actual;
+        const expected = new Error("Missing required files");
+
+        try {
+            await IntegrationConfiguration.getConfiguration("fake-file.file");
+        } catch (e) {
+            console.log(e);
+            actual = e;
+        }
+
+        assert.strictEqual(actual.name, expected.name);
+        assert.strictEqual(actual.message, expected.message);
+
+        try {
+            await IntegrationConfiguration.getConfiguration(jsonFile, "fake-file.file");
+        } catch (e) {
+            console.log(e);
+            actual = e;
+        }
+
+        assert.strictEqual(actual.name, expected.name);
+        assert.strictEqual(actual.message, expected.message);
+
+        try {
+            process.env.ASSET_COMPUTE_PRIVATE_KEY_FILE_PATH = "fake-file.file";
+            await IntegrationConfiguration.getConfiguration(jsonFile);
+        } catch (e) {
+            console.log(e);
+            actual = e;
+        }
+
+        assert.strictEqual(actual.name, expected.name);
+        assert.strictEqual(actual.message, expected.message);
+
+
+        try {
+            process.env.ASSET_COMPUTE_INTEGRATION_FILE_PATH = "fake-file.file";
+            await IntegrationConfiguration.getConfiguration();
+        } catch (e) {
+            console.log(e);
+            actual = e;
+        }
+
+        assert.strictEqual(actual.name, expected.name);
+        assert.strictEqual(actual.message, expected.message);
+    });
+
+    it('Input file is neither JSON nor YAML', async () => {
```

Comment:
- Use assert.throws to check the proper exception is thrown.
- Be careful about state, make sure that the variables you check are set up properly, especially if you run multiple attempts in the same test. Here, `actual` may have old content that render the test inconclusive because `actual` is not resetted. 
- Use sinon fake timers (sinon.clock) to control time values if you need to.